### DISCUSSION TITLE
have solve

### DISCUSSION THREAD

**Discussion by wangzhaohao** [idaholab/moose#17460] [Q&A Getting Started] [answered]
hello I am a new starter. today I meet a problem. i used the HeatConduction so i turn on the switch in the make .file. 
ALL_MODULES                 := no

CHEMICAL_REACTIONS          := no
CONTACT                     := no
EXTERNAL_PETSC_SOLVER       := no
FLUID_PROPERTIES            := no
FUNCTIONAL_EXPANSION_TOOLS  := no
GEOCHEMISTRY                := no
HEAT_CONDUCTION             := yes
LEVEL_SET                   := no
MISC                        := no
NAVIER_STOKES               := no
PHASE_FIELD                 := no
POROUS_FLOW                 := no
RAY_TRACING                 := no
RDG                         := no
RICHARDS                    := no
STOCHASTIC_TOOLS            := no
TENSOR_MECHANICS            := no
XFEM                        := no

and then creat a .i, 

[Mesh]
file = 2021329.e
[]
[Variables]
 [./temp]
 order=FIRST
 family=LAGRANGE
 initial_condition = 500.0
 [../]
[]
[Kernels]
 [./temp]
 type = HeatConduction
 variable = temp
 [../]
 [./heat_source]
 type = HeatSource
 variable = temp
 black = '3'
 scale = 5.0e8
 [../]
 [./euler]
 type = HeatConductonTimeDerivative
 variable = temp
 [../]
[]
[BCs]
 [./fuel_outside]
 type = DirichletBC
 variable = temp
 boundary = '2'
 value = 700
 [../]
[]
[Materials]
 [./thermal_pellet]
  type = HeatConductionMaterial
  block = '3'
  thermal_conductivity_temperature_function = 3.048
  specific_heat_temperature_function = 324.087
  density_temperature_function = 11078.368
  [../]
  []
[Executioner]
  type = Transient
  solve_type = 'PJFNK'
[]
[Outputs]
 execute_on = 'timestep_end'
 exodus = true
[]

then make -j8, and peacock ,but turn wrong

A 'HeatConduction' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

**Comment by cticenhour** [comment, marked as answer]
I would try a clean of the application libraries in order to verify that it's not an old library lingering around causing a registration issue. In your main application directory, run: 

```bash
make clobberall
make -j8
```

Then try running your input file again.

**Reply by aaaaaaqing** [reply]
I do ‘make clobberall’
but computer die

**Reply by aaaaaaqing** [reply]
when do ‘make -j8' and computer die

**Reply by GiudGiud** [reply]
try just “make” then 

if the computer is dying it s probably running out of memory

**Comment by aaaaaaqing** [comment]
Thank u for your help
I have a question. Currently, I would like to import data in CSV format that includes doping values at each grid point. Does MOOSE support this functionality? Additionally, I would appreciate your assistance in determining if it is possible to perform calculations between the imported doping values and test functions or solution values at specific points. Your help would be greatly appreciated




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Guillaume ***@***.***&gt;; 
发送时间: 2023年7月3日(星期一) 中午11:23
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [idaholab/moose] have solve (Discussion #17460)





 
try just “make” then
 
if the computer is dying it s probably running out of memory

**Reply by GiudGiud** [reply]
Hello

If your grid is regular, you can use this function
https://mooseframework.inl.gov/source/functions/PiecewiseMultilinear.html
If your grid is not regular, use this user object with read_mode = voronois
https://mooseframework.inl.gov/source/userobjects/PropertyReadFile.html
and this function
https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html

Guillaume

**Comment by aaaaaaqing** [comment]
I tried many ways to solve this problem, but I failed. How should this problem be solved




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "idaholab/moose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月10日(星期一) 凌晨4:01
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [idaholab/moose] have solve (Discussion #17460)





 
Hello
 
If your grid is regular, you can use this function
 https://mooseframework.inl.gov/source/functions/PiecewiseMultilinear.html
 If your grid is not regular, use this user object with read_mode = voronois
 https://mooseframework.inl.gov/source/userobjects/PropertyReadFile.html
 and this function
 https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html
 
Guillaume

**Reply by wangzhaohao** [reply]
which problem？

**Comment by aaaaaaqing** [comment]
problem in the Word


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "idaholab/moose"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月13日(星期四) 下午5:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [idaholab/moose] have solve (Discussion #17460)





 
which problem？

**Reply by GiudGiud** [reply]
open your own post please
